File: Algebra_SortFraction.vb
Module Algebra_sort_fraction

    Sub Main()
        ''
        ' Sorting fractions
        ''
        Dim numerators = {9, 3, 5, 7}
        Dim denominators = {2, 4, 12, 18}
        Dim fractions As New Dictionary(Of String, Integer())
        fractions.Add("numerators", numerators)
        fractions.Add("denominators", denominators)

        Console.WriteLine("    Sorting in ascending order the fractions:" & vbCrLf)
        ' Print as fraction
        For Each numerator In fractions.Item("numerators")
            Console.Write(String.Format("{0,13}", numerator))
        Next
        Console.Write(Environment.NewLine & String.Format("{0,12}", " "))
        For wasted = 0 To numerators.Count - 2
            Console.Write(String.Format("{0}", "-   ,        "))
        Next
        Console.WriteLine(String.Format("{0,1}", "-"))
        For Each denominator In fractions.Item("denominators")
            Console.Write(String.Format("{0,13}", denominator))
        Next
        Console.WriteLine("")

        ' use the SortFraction Class
        Dim sort_fract As New SortFraction
        sort_fract._init_(fractions)
        fractions = sort_fract.sortAscending()

        Console.WriteLine(Environment.NewLine)

        ' Print as fraction
        For Each numerator In fractions.Item("numerators")
            Console.Write(String.Format("{0,13}", numerator))
        Next
        Console.Write(Environment.NewLine & String.Format("{0,12}", "Answer  =   "))
        For wasted = 0 To numerators.Count - 2
            Console.Write(String.Format("{0}", "-   ,        "))
        Next
        Console.WriteLine(String.Format("{0,1}", "-"))
        For Each denominator In fractions.Item("denominators")
            Console.Write(String.Format("{0,13}", denominator))
        Next

    End Sub

End Module

File: Algebra_SortFraction.py
#!/usr/bin/python
from SortFraction import ArrangeFraction

##
 # Sorting fractions
 ##
numerators = [1, 3, 5, 9]
denominators = [2, 4, 2, 10]
fractions = {'numerators':numerators, 'denominators':denominators}

print("\n    Sorting in ascending order the fractions:\n")
# Print as fraction
for numerator in fractions['numerators']: print('{:12d}'.format(numerator), end='')
print('{}{:>11}'.format('\n', ' '), end='')
for wasted in range(len(numerators)-1): print('{}'.format('-   ,       '), end='')
print('{:>1}'.format('-'))
for denominator in fractions['denominators']: print('{:12d}'.format(denominator), end='') 
print('\n\n')

# use the SortFraction class
sort_fract = ArrangeFraction(fractions)
fractions = sort_fract.sortAscending()

# Print as fraction
for numerator in fractions['numerators']: print('{:13d}'.format(numerator), end='')
print('{}{:>1}'.format('\nAnswer  =  ', ' '), end='')
for wasted in range(len(numerators)-1): print('{}'.format('-   ,        '), end='')
print('{:>1}'.format('-'))
for denominator in fractions['denominators']: print('{:13d}'.format(denominator), end='') 
print('\n\n')


print('\n\n')
